### DISCUSSION TITLE
[Section 6.35 - Cognito] Lambda Layer

### DISCUSSION THREAD

**Discussion by jacksonpedrini** [chgasparoto/curso-aws-com-terraform#85] [Q&A]
Bom dia,

Ao dar um terrafor apply está me retornando abaixo:

`Plan: 2 to add, 1 to change, 1 to destroy.
null_resource.build_lambda_layers: Destroying... [id=7348244024028810713]
null_resource.build_lambda_layers: Destruction complete after 0s
null_resource.build_lambda_layers: Creating...
null_resource.build_lambda_layers: Provisioning with 'local-exec'...
null_resource.build_lambda_layers (local-exec): Executing: ["cmd" "/C" "npm install --production && cd ../ && zip -9 -r --quiet joi.zip *"]
null_resource.build_lambda_layers (local-exec): npm WARN serverless-app@1.0.0 No repository field.

null_resource.build_lambda_layers (local-exec): audited 6 packages in 0.547s
null_resource.build_lambda_layers (local-exec): found 0 vulnerabilities

null_resource.build_lambda_layers (local-exec): 'zip' is not recognized as an internal or external command,
null_resource.build_lambda_layers (local-exec): operable program or batch file.
╷
│ Error: local-exec provisioner error
│
│   with null_resource.build_lambda_layers,
│   on lambda.tf line 6, in resource "null_resource" "build_lambda_layers":
│    6:   provisioner "local-exec" {
│
│ Error running command 'npm install --production && cd ../ && zip -9 -r --quiet joi.zip *': exit status 1. Output: npm WARN
│ serverless-app@1.0.0 No repository field.
│
│ audited 6 packages in 0.547s
│ found 0 vulnerabilities
│
│ 'zip' is not recognized as an internal or external command,
│ operable program or batch file.`

@chgasparoto pode me dar um help?

**Comment by chgasparoto** [comment]
@jacksonpedrini qual seu sistema operacional? O erro está falando que você não tem o software **_zip_** instalado.

**Comment by jacksonpedrini** [comment]
@chgasparoto Windows 11, acabei de instalar o tinha o Winrar, então acabei de instalar o 7z e setei nas variáveis de Path do SO. Só tive que mudar os parâmetros no código  de "/ && zip -9 -r --quiet ${local.layer_name} *" para  "/ && zip a ${local.layer_name} *"".

**Reply by chgasparoto** [reply]
show de bola!
